src/cluster_legacy.c

```diff
@@ -6890,9 +6939,26 @@ void clusterPromoteSelfToMaster(void) {
     replicationUnsetMaster();
 }
 
+void updateAllCachedNodesHealth(void) {
```

Comment:
I think this function name can be improved to be more clear about the action - it is not just an "update" but `invalidateSlotCommandCacheOnNodeHealthChange` or `invalidateTopologyCacheOnNodeHealthChange` if we anticipate more cache like this can be introduced.

Reply:
This function only detects the change and updates the `node->is_node_healthy`. This can be safely called later as well, if we want to do more caching in the future.

Reply:
And then how about `updateAndDetectAllCachedNodesHealth()`?

Reply:
`detectAndUpdateChangeOnNodeHealthChange`